backend/src/graphql_api/plt.rs

```diff
@@ -489,14 +489,69 @@ impl QueryPLTAccountAmount {
             .map_err(|e| ApiError::InternalServerError(InternalError::InternalError(e.to_string())))
     }
 
-    async fn plt_accounts_by_token_id(
+    async fn plt_accounts_by_token_id<'a>(
         &self,
-        ctx: &Context<'_>,
-        token_id: types::ID,
-    ) -> ApiResult<Vec<PLTAccountAmount>> {
+        ctx: &Context<'a>,
+        id: types::ID,
+        #[graphql(desc = "Returns the first _n_ elements from the list.")] first: Option<u64>,
+        #[graphql(desc = "Returns the elements in the list that come after the specified cursor.")]
+        after: Option<String>,
+        #[graphql(desc = "Returns the last _n_ elements from the list.")] last: Option<u64>,
+        #[graphql(desc = "Returns the elements in the list that come before the specified cursor.")]
+        before: Option<String>,
+    ) -> ApiResult<connection::Connection<String, PLTAccountAmount>> {
         let pool = get_pool(ctx)?;
-        let token_id: TokenId = token_id.to_string();
-        Ok(PLTAccountAmount::query_by_token_id(pool, token_id).await?)
+        let token_id = TokenId::from_str(id.as_ref()).map_err(|e| {
+            ApiError::InternalServerError(InternalError::InternalError(format!(
+                "Failed to parse token ID: {}",
+                e
+            )))
+        })?;
+        let config = get_config(ctx)?;
+        let query = ConnectionQuery::<DescendingI64>::new(
+            first,
+            after,
+            last,
+            before,
+            config.plt_token_events_collection_limit,
```

Comment:
This query seems not related to `token_events`, so either rename that variable or use a separate variable.